dmoj/executors/mixins.py

```diff
@@ -10,158 +10,117 @@
+            name = self.get_executor_name()
+            fs = BASE_FILESYSTEM + self.fs + env.get('extra_fs', {}).get(name, []) + [re.escape(self._dir)]
+            return fs
+
+        def get_write_fs(self):
+            return BASE_WRITE_FILESYSTEM + self.write_fs
+
+        def get_allowed_syscalls(self):
+            return self.syscalls
+
+        def get_address_grace(self):
+            return self.address_grace
+
+        def get_env(self):
+            env = {'LANG': 'C.UTF-8'}
+            if self.unbuffered:
+                env['CPTBOX_STDOUT_BUFFER_SIZE'] = 0
+            return env
+
+        def launch(self, *args, **kwargs):
+            for src, dst in kwargs.get('symlinks', {}).items():
+                src = os.path.abspath(os.path.join(self._dir, src))
+                # Disallow the creation of symlinks outside the submission directory.
+                if os.path.commonprefix([src, self._dir]) == self._dir:
+                    # If a link already exists under this name, it's probably from a
+                    # previous case, but might point to something different.
+                    if os.path.islink(src):
+                        os.unlink(src)
+                    os.symlink(dst, src)
+                else:
+                    raise InternalError('cannot symlink outside of submission directory')
+
+            agent = self._file('setbufsize.so')
+            shutil.copyfile(setbufsize_path, agent)
+            env = {
+                # Forward LD_LIBRARY_PATH for systems (e.g. Android Termux) that require
+                # it to find shared libraries
+                'LD_LIBRARY_PATH': os.environ.get('LD_LIBRARY_PATH', ''),
+                'LD_PRELOAD': agent,
+                'CPTBOX_STDOUT_BUFFER_SIZE': kwargs.get('stdout_buffer_size'),
+                'CPTBOX_STDERR_BUFFER_SIZE': kwargs.get('stderr_buffer_size'),
+            }
+            env.update(self.get_env())
+
+            return SecurePopen([utf8bytes(a) for a in self.get_cmdline() + list(args)],
+                               executable=utf8bytes(self.get_executable()),
+                               security=self.get_security(launch_kwargs=kwargs),
+                               address_grace=self.get_address_grace(),
+                               data_grace=self.data_grace,
+                               personality=self.personality,
+                               fds=kwargs.get('fds'),
+                               time=kwargs.get('time'),
+                               memory=kwargs.get('memory'),
+                               wall_time=kwargs.get('wall_time'),
+                               stdin=kwargs.get('stdin'),
+                               stdout=kwargs.get('stdout'),
+                               stderr=kwargs.get('stderr'),
+                               env=env, cwd=utf8bytes(self._dir),
+                               nproc=self.get_nproc(),
+                               fsize=self.fsize)
 except ImportError:
```

Comment:
Why are we still doing this? Let's just have the *nix `PlatformExecutorMixin` directly in the global scope. Then if things fail, you'd know what `cptbox` error it is instead of `NameError` on `PlatformExecutorMixin`.